plugins/lockToVote/hooks/useProposalVeto.tsx

```diff
@@ -73,12 +104,58 @@ export function useProposalVeto(proposalId: string) {
   }, [vetoingStatus, vetoTxHash, isConfirming, isConfirmed]);
 
   const vetoProposal = () => {
-    vetoWrite({
-      abi: LockToVetoPluginAbi,
-      address: PUB_LOCK_TO_VOTE_PLUGIN_ADDRESS,
-      functionName: "veto",
-      args: [proposalId, 50000000000000000000],
-    });
+    if (!proposal || !balanceResult || !nonceResult || !nameResult || !versionResult) return;
+    const balance = BigInt(Number(balanceResult?.result));
+    const nonce = BigInt(Number(nonceResult?.result));
+    const erc20_name = nameResult?.result;
+    const versionFromContract = versionResult?.result;
+
+    let deadline = BigInt(Math.floor(Date.now() / 1000) + 60 * 60);
+    let value = balance; 
+    let destination: `0x${string}` = PUB_LOCK_TO_VOTE_PLUGIN_ADDRESS;
+
+    const domain = {
+      chainId: PUB_CHAIN.id,
+      name: String(erc20_name),
+      /* We assume 1 if permit version is not specified */
+      version: String(versionFromContract ?? '1'),
+      verifyingContract: PUB_TOKEN_ADDRESS,
+    };
+
+    const types = {
+      Permit: [
+        { name: 'owner', type: 'address' },
+        { name: 'spender', type: 'address' },
+        { name: 'value', type: 'uint256' },
+        { name: 'nonce', type: 'uint256' },
+        { name: 'deadline', type: 'uint256' },
+      ],
+    };
+
+    const message = {
+      owner: account_address,
+      spender: destination,
+      value,
+      nonce,
+      deadline,
+    };
+
+    vetoPermitSign({
```

Comment:
This part, plus the message could be in their own useHook at the root of Aragonette (not inside this plugin)